app/models/user.rb

```diff
@@ -12,6 +12,14 @@ def last_ten
     destinations.order(created_at: :desc).limit(10)
   end
 
+  def self.top_carbon_savers(limit = 10)
+    User.joins(:trip_logs).select("users.*, sum(trip_logs.miles) as carbon_saved").group("users.id").order("carbon_saved desc").limit(limit)
+  end
+
```

Comment:
This could be refactored to be going off of the `trip_logs` table directly and not joining from `users`. The same situation as the optimization dojo. Saves a couple 100 ms.